src/SPC/builder/extension/swoole.php

```diff
@@ -49,7 +49,9 @@ public function getUnixConfigureArg(): string
 
         // additional feature: c-ares, brotli, nghttp2 (can be disabled, but we enable it by default in config to support full network feature)
         $arg .= $this->builder->getLib('libcares') ? ' --enable-cares' : '';
-        $arg .= $this->builder->getLib('brotli') ? (' --with-brotli-dir=' . BUILD_ROOT_PATH) : '';
+        if (!$this->isBuildShared()) {
+            $arg .= $this->builder->getLib('brotli') ? (' --enable-brotli --with-brotli-dir=' . BUILD_ROOT_PATH) : '';
+        }
```

Comment:
why can't it be enabled if the shared build is also enabled?

Reply:
When I use a shared build, using `buildroot/bin/php -d "extension=swoole.so" --ri swoole`, it will show missing symbols when loading the brotli version number. But if I don't add this line to the build, it will display normally, which is very strange.

Reply:
>the check here should be whether we're currently building static or shared

Now that we have it like this, `isBuildShared()` returns true only when `--with-dynamic` includes swoole.

Reply:
> "swoole" --with-shared="swoole"

We always assume someone want to build static php cli (or fpm, etc.), and dynamic extension is used for it. I don't think it's a good idea to allow building both static and dynamic versions at the same time. At least we have not added default sanity check for dynamic extension yet, which might affect it.

Reply:
@crazywhalecc I think we should keep it the way it is with one build command that adds `--build-shared`  to *also* build the shared library. Not a either or.

Reply:
The use case would be something like building frankenphp + xdebug + ffi shared in one go. When an extension can be built as both static and shared, the best way with the fewest spaghetti checks would be to simply build both, rather than make an assumption about the users intent or error out.

 But if you wish to separate them into two different commands, that's fine too.

Reply:
We do want to avoid blaming our own problems on user input errors, but at the same time, the build not only requires changing the current build process, but also makes it possible for the sanity check to fail to verify the availability of shared extensions (such as this swoole issue). So I think it would be best if we add a command to build the extension, but we would make it in another PR.

Reply:
To clarify the results:

1. spc allows building `bin/spc build "" --build-shared=swoole,xdebug`
2. spc allows passing in a combination like `bin/spc build swoole --build-shared=swoole`, but the final built CLI (or other SAPI executable) will not contain swoole, but we will get `swoole.so` in `lib`.

Reply:
> We should refactor it out to its own command rather than make an arbitrary decision about what the user meant.

You mean we build shared extension totally separated in another command?

Reply:
Yes. Arbitrarily deciding that a user meant `bin/spc build "" --build-shared=swoole` or `bin/spc build "swoole" --build-shared=""` when they type `bin/spc build swoole --build-shared=swoole` is just a bad idea.

So either we build static and shared at the same time, or we refactor shared extension building out to its own command.